src/api/mod.rs

```diff
@@ -403,6 +451,119 @@ impl<'a> ValueListBuilder<'a> {
     }
 }
 
+//TODO: move to its own module metadata, with related types
```

Comment:
I don't have strong opinions for this PR: either leaving it as is or creating a `meta.rs` module is fine with me. Eventually I'd like to move what's everything in here to separate modules but you have free reign in this PR.